tools/ci-build/README.md

```diff
@@ -0,0 +1,39 @@
+ci-build
+========
+
+This directory includes the build Docker image and scripts to use it in CI.
+- `acquire-build-image`: Script that retrieves the build image from public ECR or builds one locally
+  depending on the state of the tools directory. If the tools have changed (in git history), then it opts
+  to build a new image rather than reuse an existing one.
+- `add-local-user.dockerfile`: Creates a user in the build image with the host's user ID
+- `build-image.dockerfile`: Dockerfile used to create the build image
+- `build.docker-compose.yml`: Docker Compose file for using the build image
+- `ci-action`: Script for running CI actions inside of the Docker build image
+- `ci-create-workspace`: Used by `ci-action`, but can be run manually to create a one-off workspace for debugging
+- `scripts/`: CI scripts that get copied into the build image
+
+There are three spaces you need to conceptualize for testing this locally:
+- **Origin:** The original `smithy-rs` where you're iterating on CI scripts.
+- **Starting space:** Directory with `smithy-rs` to run CI checks against. You have to create this. Conceptually,
+  this is equivalent to the GitHub Actions working directory.
+- **Action space:** Temporary directory maintained by `ci-action` where the CI checks actually run.
+
+To create the starting space, do the following:
+
+```bash
+cd /path/to/my/starting-space
+git clone https://github.com/awslabs/smithy-rs.git
+# Optionally check out the revision you want to work with in the checked out smithy-rs.
+# Just make sure you are in /path/to/my/starting-space (or whatever you called it) after.
```

Comment:
seems like we might eventually want to automate this to provide testing directly against a specified githash or tag